### PR TITLE
Tests/initialize basic tests

### PR DESCRIPTION
**Pull request by Blent1050**
**Make sure to download the new dependancies after pulling in this PR**

## This feature introduces basic tests to a few components in the application
The coverage contains:
- Landing Page Components
- Navigation Bar Components

**These are basic tests to get us up and running, I left some notes on some tests we may want to add on in the future**.

## Utils
I also added two utils for our tests, `mockStore` & `mockAuth`.

We can leverage these utils in tests that require our store/auth components. An example of their usage can be found inside of `heroSection.test.jsx`. 

**Note**: When using the mockAuth component, you will need to mock the auth0 wrapper at the top of the file after the test imports. 
Ex: `jest.mock('../../../auth/react-auth0-wrapper.jsx');`

## File Structure of Tests
I created a folder in `src` called tests. These will house our various tests. 
The building blocks of the folder are:
- `Module folder`
    - This folder essentially mimics the folder of the module. Any files within the module folder would contain a single test file that would store all of our tests. (We can change this easily with how it's currently implemented).
- `util`
    - Contains our various utils we setup for tests. 
- `global tests - such as app.test.jsx`

## Cypress
I initialized cypress with our app to cover E2E testing. You can check it out by running `yarn cypress`. Simply click on a test to run that test when the test runner opens. 
- I didn't write any cypress tests yet, just a basic example test. 
- circleCI will need to be properly setup to run cypress in their docker container. I'm not sure if the current circleCI script will work. I'll need additional testing after this PR is merged to check if it is properly setup.

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "projectId": "x8gohh"
```
Comment (Alex-AK) [inline cypress.json:2]: Does this need to be an env variable?

File: src/store/actions/userActions.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ export const getUserById = (
   dispatch({ type: GET_USER_DATA_START });
 
   try {
-    const userData = await Axios.get(`/users/${userId}`, {
+    await Axios.get(`/users/${userId}`, {
       headers: { Authorization: token },
```
Comment (Alex-AK) [inline src/store/actions/userActions.js:53]: The authorization token is attached in the axios config. We can remove this.

File: src/tests/Navigation/navigation.test.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import MobileMenu from '../../modules/Navigation/MobileMenu';
+import MobileMenuButton from '../../modules/Navigation/MobileMenuButton';
+import Navbar from '../../modules/Navigation/Navbar';
+// import NavLinks from '../../modules/Navigation/NavLinks';
+
+// utils
+import mockStore from '../util/mockStore';
+import mockAuth from '../util/mockAuth';
+
+jest.mock('../../auth/react-auth0-wrapper.jsx');
```
Comment (Alex-AK) [inline src/tests/Navigation/navigation.test.jsx:16]: Do we need to do jest.mock auth when using the mockAuth utility function on beforeEach()?

File: cypress.json (hunk 1 above)
Comment (Blent1050) [reply cypress.json:2]: Cypress doesn't mention this is sensitive and I can't find anything about it. This just references our project suite, so if someone got a hold of it I'm not sure if they can really do any damage.

File: src/tests/Navigation/navigation.test.jsx (hunk 3 above)
Comment (Blent1050) [reply src/tests/Navigation/navigation.test.jsx:16]: Should just have to mock it once.

**Comment by Blent1050** [comment]
@Alex-AK  I think we're good to merge this if you don't have anything else to add.

**Pull request merged**